cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/GlobalParam.scala

```diff
@@ -0,0 +1,213 @@
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalKey[T]
+sealed trait GlobalParamKey[T] extends GlobalKey[T]
```

Comment:
what does the two different classes here do as opposed to 1?

Reply:
Nothing! As long as we have separation of Params and ServiceParams somewhere (so in this case in the different dictionaries) - we're good. Got rid of the two keys